plugins/woocommerce-blocks/assets/js/data/cart/index.ts

```diff
@@ -32,8 +39,29 @@ const registeredStore = registerStore< State >( STORE_KEY, {
 	__experimentalUseThunks: true,
 } );
 
+register( store );
+
+declare module '@wordpress/data' {
+	function dispatch(
+		key: typeof STORE_KEY
+	): DispatchFromMap< typeof actions >;
+	function select( key: typeof STORE_KEY ): SelectFromMap<
+		typeof selectors
+	> & {
+		hasFinishedResolution: ( selector: string ) => boolean;
+	};
+}
+
+// The resolver for getCartData fires off an API request. But if we know the cart is empty, we can skip the request.
+// The only reliable way to check if the cart is empty is to check the cookies.
+window.addEventListener( 'load', () => {
+	if ( ! hasCartSession() ) {
+		wpDispatch( STORE_KEY ).finishResolution( 'getCartData' );
+	}
```

Comment:
This feels like a hacky approach, so open to suggestions for how to tackle this without the addEventListener.